bazel/re2c.bzl

```diff
@@ -0,0 +1,81 @@
+# re2c generator rules - assuming that re2c is available as '@re2c//:re2c'.
+
+visibility("public")
+
+def _re2c_impl(ctx):
+    srcs = ctx.files.srcs
+    if len(srcs) != 1:
+        fail("Attribute `srcs` must contain exactly 1 '.re2c' file.")
+    outs = ctx.outputs.outs
+    if not outs or [o for o in outs if o.path.split(".")[-1] not in ["c", "cc", "h"]]:
+        fail("Attribute `outs` must contain at least 1 '.cc' or '.h' file.")
+    err_file = ctx.actions.declare_file(ctx.attr.name + ".err")
+    ctx.actions.run_shell(
+        outputs = outs + [err_file],
+        tools = [ctx.executable._re2c_tool],
+        inputs = ctx.files.srcs + ctx.files.data,
+        command = """
+            line_args=""
+            [ -z "${re2c_parse_first_line}" ] && \\
+                line_args="$(sed -rne "1s,// re2c .INPUT -o .OUTPUT (.+),\\1,p" "{src}")"
+            "{re2c}" "{src}" -o "{out}" ${{line_args}} {re2c_args} {includes} > "{err}" 2>&1
+            cat "{err}" >> "{out}"
```

Comment:
Why do you append re2c error messages / warnings to the output file? We do this for tests to make diffing simpler, but I wouldn't expect it in a normal re2c invocation. I'd rather just let them be without redirecting to `{err}` file.

By the way, where does the `{err}` file go, is it just left in the build directory?

Reply:
The error file stays in bazel's gendir output directory. It serves no further use at this point. In theory we could do the same with a temp file, but it is better to let bazel decide where the file goes. And once it is generated, it might we well be kept in place.

In any event it is likely better to do this already in the tool if needed. That will also make it easy to issue "#error" lines.